src/gui/systray.cpp

```diff
@@ -280,7 +280,7 @@ QRect Systray::currentScreenRect() const
 {
     const auto screen = currentScreen();
     const auto rect = screen->geometry();
-    return rect.translated(screen->virtualGeometry().topLeft());
+    return rect;
```

Comment:
Probably nitpicking, but I would just `return currentScreen()->geometry();` and maybe even make this function inline.

Reply:
Actually I've been pondering changing the return, didn't do it as mainly cosmetic but why not.

The inline I'm not fond of it. It's a non-binding hint to the compiler which will decide to inline or not... even if the keyword is not here. And this is best case scenario, some compilers (used to, and hopefully improved,) produce less efficient code with use of inline than without (because it was skewing their optimizer heuristics).
AFAIK in C++17 definition of the keyword even got changed due to that to relax the meaning to "multiple definitions are permitted" (which is more aligned with the "might inline or not" behavior we get in practice).

Reply:
> didn't do it as mainly cosmetic but why not.

I still don't understand why you keep the `screen` variable. I don't see any benefit.

> The inline I'm not fond of it. It's a non-binding hint to the compiler which will decide to inline or not...

Oh, I had to be more clear. I wasn't talking about adding an `inline` keyword (although I would probably do this), but about moving the trivial `return currentScreen()->geometry();` function body to the body of the class definition. Otherwise the compiler won't ever inline unless you use link-time optimization (which we don't by default).

Reply:
> > didn't do it as mainly cosmetic but why not.
> 
> I still don't understand why you keep the `screen` variable. I don't see any benefit.

Because I wanted to assert on it and then forgot (twice!) to add it. Adding the missing assert.
 
> > The inline I'm not fond of it. It's a non-binding hint to the compiler which will decide to inline or not...
> 
> Oh, I had to be more clear. I wasn't talking about adding an `inline` keyword (although I would probably do this), but about moving the trivial `return currentScreen()->geometry();` function body to the body of the class definition. Otherwise the compiler won't ever inline unless you use link-time optimization (which we don't by default).
> 
> **edit**: This inlining is a thing that I would do and I commented this to let you ask yourself why you don't do it – there are valid reasons, I don't judge. That's why I only wrote a comment and not a review that requested changes :-)

Well no then. :-)

Moving to the header would force pulling QScreen definition in the Systray definition. I'd rather not bind them even more. It's a something I would do when there's a strong performance impact but here we're talking about a handful of calls at most on a single user interaction, it's negligible compared to the rest of the computation happening and model updates in the GUI.